tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,339 @@
+LOGEDOFF_MSG = "You are currently not logged in"
+
+
+def configure_sessions(enable='true'):
+    """Enables the `use_sessions` option in hammer config"""
+    ssh.command(
+        '''sed -i -e '/username/d;/password/d;/use_sessions/d' {0};\
+        echo '  :use_sessions: {1}' >> {0}'''.format(
+            HAMMER_CONFIG, enable
+        )
+    )
+
+
+@run_in_one_thread
+class HammerAuthTestCase(CLITestCase):
+    """Implements hammer authentication tests in CLI"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates users to be reused across tests"""
+        super(HammerAuthTestCase, cls).setUpClass()
+        cls.uname_admin = gen_string('alpha')
+        cls.uname_viewer = gen_string('alpha')
+        cls.password = gen_string('alpha')
+        cls.mail = 'test@example.com'
+        User.create({
+                'auth-source-id': '1',
+                'login': cls.uname_admin,
+                'password': cls.password,
+                'mail': cls.mail,
+                'admin': '1',
+        })
+        User.create({
+                'auth-source-id': '1',
+                'login': cls.uname_viewer,
+                'mail': cls.mail,
+                'password': cls.password,
+        })
+        User.add_role({'login': cls.uname_viewer, 'role': 'Viewer'})
+
+    @tier1
+    def test_positive_create_session(self):
+        """Check if user stays authenticated with session enabled
+
+        :id: fcee7f5f-1040-41a9-bf17-6d0c24a93e22
+
+        :Steps:
+
+            1. Set use_sessions, set short expiration time
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Wait until session expires, assert credentials
+               are required
+
+        :expectedresults: The session is successfully created and
+            expires after specified time
+        """
+        try:
+            configure_sessions()
+            Settings.set({'name': 'idle_timeout', 'value': 1})
+            Auth.login({
+                'username': self.uname_admin,
+                'password': self.password
+            })
+            result = Auth.status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            # list organizations without supplying credentials
+            orglist = Org.list(credentials=False, return_raw_response=True)
+            self.assertEqual(orglist.return_code, 0)
+            # wait until session expires
+            sleep(70)
+            orglist = Org.list(credentials=False, return_raw_response=True)
+            self.assertNotEqual(orglist.return_code, 0)
+            result = Auth.status()
+            self.assertIn(
+                LOGEDOFF_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+        finally:
+            # reset timeout to default
+            Settings.set({'name': 'idle_timeout', 'value': 60})
```

Comment:
nitpick: what if specific satellite box used different setting value for any reason? I'd rather cache the value before the test and reset back to that value.